src/components/Brushbox-textures.tsx

```diff
@@ -1,37 +1,47 @@
 import { useState } from "react";
 import './Brushbox-textures.css';
 import { Terrain } from "../Model/terrain";
+import { BrushSelection } from "../context";
 
-const BrushBoxTextures = ({ setBrushSelection }) => {
-    const [clicked, setClicked] = useState(-1)
-    const [key, setKey] = useState(0)
+interface BrushBoxProps {
+  setBrushSelection: (selection: BrushSelection) => void;
+  brushSelection: BrushSelection;
+}
 
-    const onClickIcon = (terrain: Terrain) => {
-        setClicked(terrain)
-        setBrushSelection({ terrain: terrain })
-    }
+const BrushBoxTextures = (props: BrushBoxProps) => {
+  const { setBrushSelection, brushSelection } = props
 
-    const selected = clicked == key ? "selected" : "";
+  const onClickIcon = (terrain: Terrain) => {
+    setBrushSelection({ terrain });
+  }
 
-    return (
-        <div className='icon-box' >
-            <div className={`icon-button ${clicked == 0 ? "selected" : ""}`} onClick={() => onClickIcon(0)}>
-                <div className='brushbox-texture' >
-                    default
-                </div>
-            </div>
-            <div id='2' className={`icon-button ${clicked == 1 ? "selected" : ""} water`} onClick={() => onClickIcon(1)}>
-                <div className='brushbox-texture water' >
-                    water
-                </div>
-            </div>
-            <div className={`icon-button ${clicked == 2 ? "selected" : ""} mountain`} onClick={() => onClickIcon(2)}>
-                <div className='brushbox-texture mountain' >
-                    mountain
-                </div>
-            </div>
+  return (
+    <div className='icon-box' >
+      <div
+        className={`icon-button ${brushSelection.terrain === Terrain.NORMAL ? "selected" : ""}`}
```

Comment:
this is fine, but I think the industry standard way of dealing with this is to [classnames](https://www.npmjs.com/package/classnames#usage-with-reactjs). 

See https://legacy.reactjs.org/docs/faq-styling.html